tidy3d/components/source/freq_range.py

```diff
@@ -0,0 +1,202 @@
+""" Utility class `FreqRange` for frequency and wavelength handling."""
+
+from __future__ import annotations
+
+
+import numpy as np
+import pydantic.v1 as pydantic
+
+from tidy3d import constants as td_const  
+from tidy3d.components.source.time import GaussianPulse
+from tidy3d.components.base import Tidy3dBaseModel
+
+class FreqRange(Tidy3dBaseModel):
+
+    freq0: pydantic.PositiveFloat = pydantic.Field(
+        1.0, title="Central frequency", description="Real-valued positive central frequency", units="Hz")
+    
+    fwidth: pydantic.PositiveFloat = pydantic.Field(
+        1.0, title="Frequency bandwidth", description="Real-valued positive width of the frequency range (bandwidth)", units="Hz") 
+    """
+        Convenience class for handling frequency/wavelength conversion and assignment
+
+        Example
+        -------
+        >>> freq0  = 1e12
+        >>> fwidth = 1e11
+        >>> freq_range = FreqRange(freq0=freq0, fwidth=fwidth)
```

Comment:
I think it would be helpful to also add a few lines to the example showing off how you can use `freq_range`. Like very simply calling some of the properties, `freq_range.freqs` for example, or `freq_range.to_gaussian_pulse()`